addons/common/XEH_postInit.sqf

```diff
@@ -251,6 +251,35 @@ if (_currentVersion != _previousVersion) then {
 
 call FUNC(checkFiles);
 
+// Let the server know that we have ACE loaded
+missionNamespace setVariable [QGVAR(aceLoaded_) + getPlayerUID player, true, 2];
+
+// This handles clients joining that do not have ACE loaded
+if (isServer) then {
+    addMissionEventHandler ["PlayerConnected", {
+        [{
+            params ["", "_uid", "", "", "_owner"];
+
+            // If ACE is not loaded, kick unit
+            if !(missionNamespace getVariable [QGVAR(aceLoaded_) + _uid, false]) then {
+                // Send function to client
+                _owner publicVariableClient QFUNC(errorMessage);
```

Comment:
Neat.

Reply:
The only problem is that, as you pointed out in your deleted comment, `missionNamespace getVariable [QGVAR(aceLoaded_) + _uid, false]` will not be set by clients who have an outdated ACE.

The easiest fix would be to change the error message on [line 270](https://github.com/acemod/ACE3/pull/9266/files#diff-d241a0f897acd01a4b09998a038b0c1afd584435e7ae84d08c50755d9b914ce1R270) to `"ACE is not present or outdated"`, but if we want to fix it properly, I'm not sure how to go about it.

Reply:
Add build number to variable name and update the error message?

Reply:
> Add build number to variable name and update the error message?

I don't think I was clear with my explanation, as I don't understand your suggestion. By "outdated" I meant a version of ACE that does not support this feature, meaning it does not run `missionNamespace setVariable [QGVAR(aceLoaded_) + getPlayerUID player, true, 2]`.

How would adding the build number help in this specific case? Clients that don't have ACE loaded at all and clients with "outdated" ACE versions are indistinguishable as of the current implementation. I was wondering if there is something built into ACE that we could leverage, something that makes clients send information to the server upon connection.

Reply:
Just change the error message. We don't need to distinguish being outdated past the point of this PR or not having the mod.